src/vm/castcache.cpp

```diff
@@ -0,0 +1,291 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+//
+// File: castcache.cpp
+//
+
+#include "common.h"
+#include "castcache.h"
+
+#if !defined(DACCESS_COMPILE) && !defined(CROSSGEN_COMPILE)
+
+OBJECTHANDLE CastCache::s_cache = NULL;
+
+BASEARRAYREF CastCache::CreateCastCache(DWORD size)
+{
+    CONTRACTL
+    {
+        THROWS;
+        GC_TRIGGERS;
+        MODE_COOPERATIVE;
+    }
+    CONTRACTL_END;
+
+    // size must be positive
+    _ASSERTE(size > 0);
+    // size must be a power of two
+    _ASSERTE((size & (size - 1)) == 0);
+
+    BASEARRAYREF table = NULL;
+
+    try
```

Comment:
CoreCLR has to use EX_TRY/EX_CATCH macros. C++ exceptions cannot be used directly.

Reply:
will change to   EX_TRY/EX_CATCH   

---
In reply to: [335154190](https://github.com/dotnet/coreclr/pull/23548#discussion_r335154190) [](ancestors = 335154190)